File: py/pe16.py
#! /usr/bin/env python3

from sys import argv
from pe import sod

def pe16(n):
    """
    What is the sum of the digits of the number 2**1000?
    """
    print("sum of the digits of 2**%d is %d" % (n, sod(2**n)))

def main():
    argc = len(argv)
    if argc > 1:
        try:
            n = int(argv[1])
        except ValueError:
            print("  python pe16.py [integer]")
            return
    else:
        n = 1000
    
    pe16(n)

if __name__ == "__main__":
    main()

File: cpp/cli/src/pe16.cc
#include "../include/pe16.h"

#include <cstdlib>
#include <iostream>

int pe16(int argc, char **argv)
{
  using namespace std;

  uint32_t n = 1000;
  if (argc > 1) {
    char *end;
    n = strtoul(argv[1], &end, 10);
  }

  cout << "n = " << n << endl;
  cout << "sum of digits = " << pe::pe16(n) << endl;

  return 0;
}
